pcdsdevices/version.py

```diff
@@ -0,0 +1,48 @@
+"""
+Experimental version handling.
+1. Pick the version for use in __init__.py and cli.py without polluting the namespace.
+2. Defer evaluation of the version until it is checked to save 0.3s on import
+3. Use the git version in a git checkout or archive and _version otherwise.
+"""
+from collections import UserString
+from pathlib import Path
+from typing import Optional
+
+
+class VersionProxy(UserString):
+    def __init__(self):
+        self._version = None
+
+    def _get_version(self) -> Optional[str]:
+        # Checking for directory is faster than failing out of get_version
+        repo_root = Path(__file__).resolve().parent.parent
+        if (repo_root / ".git").exists() or (repo_root / ".git_archival.txt").exists():
```

Comment:
I was at some point worried that this would slow down the lookup, but after thinking harder about it I think there is no case where the `.git_archival.txt` exists, is not subbed, and the git directory is missing. The tarball won't have the `.git` directory and and the install won't have the `.git_archival.txt` file or the `.git` directory.